packages/manifold/contracts/edition/ManifoldERC721Edition.sol

```diff
@@ -47,83 +48,76 @@ contract ManifoldERC721Edition is CreatorExtension, ICreatorExtensionTokenURI, I
+        _maxSupply[instanceId] = maxSupply_;
+        _tokenPrefix[instanceId] = prefix;
+        _creatorInstanceIds[creator].push(instanceId);
+        emit SeriesCreated(msg.sender, creator, instanceId, maxSupply_);
+        return instanceId;
     }
 
     /**
      * See {IManifoldERC721Edition-setTokenURIPrefix}.
      */
-    function setTokenURIPrefix(address creator, uint256 series, string calldata prefix) external override creatorAdminRequired(creator) {
-        require(series > 0 && series <= _currentSeries[creator], "Invalid series");
-        _tokenPrefix[creator][series] = prefix;
+    function setTokenURIPrefix(address creator, uint256 instanceId, string calldata prefix) external override creatorAdminRequired(creator) {
+        require(instanceId > 0, "Invalid instanceId");
+        _tokenPrefix[instanceId] = prefix;
     }
     
     /**
      * @dev See {ICreatorExtensionTokenURI-tokenURI}.
      */
     function tokenURI(address creator, uint256 tokenId) external view override returns (string memory) {
-        (uint256 series, uint256 index) = _tokenSeriesAndIndex(creator, tokenId);
-        return string(abi.encodePacked(_tokenPrefix[creator][series], (index+1).toString()));
+        (uint256 instanceId, uint256 index) = _tokenInstanceAndIndex(creator, tokenId);
+        return string(abi.encodePacked(_tokenPrefix[instanceId], (index+1).toString()));
     }
     
     /**
      * @dev See {IManifoldERC721Edition-mint}.
      */
-    function mint(address creator, uint256 series, address recipient, uint16 count) external override nonReentrant creatorAdminRequired(creator) {
+    function mint(address creator, uint256 instanceId, address recipient, uint16 count) external override nonReentrant creatorAdminRequired(creator) {
         require(count > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+count <= _maxSupply[creator][series], "Too many requested");
+        require(_totalSupply[instanceId]+count <= _maxSupply[instanceId], "Too many requested");
         
         uint256[] memory tokenIds = IERC721CreatorCore(creator).mintExtensionBatch(recipient, count);
-        _updateIndexRanges(creator, series, tokenIds[0], count);
+        _updateIndexRanges(instanceId, tokenIds[0], count);
     }
 
     /**
      * @dev See {IManifoldERC721Edition-mint}.
      */
-    function mint(address creator, uint256 series, address[] calldata recipients) external override nonReentrant creatorAdminRequired(creator) {
+    function mint(address creator, uint256 instanceId, address[] calldata recipients) external override nonReentrant creatorAdminRequired(creator) {
         require(recipients.length > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+recipients.length <= _maxSupply[creator][series], "Too many requested");
+        require(_totalSupply[instanceId]+recipients.length <= _maxSupply[instanceId], "Too many requested");
```

Comment:
Total supply needs to be creator/instanceId combo as key.  Otherwise ppl could theoretically fake mint by guessing the right instance id's

Reply:
How so - doesn't the `creatorAdminRequired` modifier prevent this?

Reply:
No it doesn't.  You are keying everything off of instanceId.

So I could instaitate it with MY contract instanceId your future instanceId, and then your frontend app could think it's already instantiated for your creator contract later, because there's no way to differentiate it because your mapping key doesn't contain creator address.

Reply:
Should be good now